src/constraint/spec.ts

```diff
@@ -323,7 +324,14 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     if (mark === Mark.AREA || mark === Mark.BAR) {
       for (let encQ of encodings) {
         if((encQ.channel === Channel.X || encQ.channel === Channel.Y) && encQ.scale) {
-          if (((encQ.scale as ScaleQuery).type as ScaleType) === ScaleType.LOG) {
+
+          const scale: ScaleQuery = encQ.scale as ScaleQuery;
+          let sType = scale.type as ScaleType;
```

Comment:
I don't think as need declarations for `sType, timeUnit, type` here as we only use them once.  Just pass them to the method.